server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfilesWs.java

```diff
@@ -28,13 +28,16 @@
   private final QProfileRestoreBuiltInAction resetAction;
   private final RuleActivationActions ruleActivationActions;
   private final BulkRuleActivationActions bulkRuleActivationActions;
+  private final ProjectAssociationActions projectAssociationActions;
 
   public QProfilesWs(QProfileRestoreBuiltInAction resetAction,
                      RuleActivationActions ruleActivationActions,
-                     BulkRuleActivationActions bulkRuleActivationActions) {
+                     BulkRuleActivationActions bulkRuleActivationActions,
+                     ProjectAssociationActions projectAssociationActions) {
```

Comment:
Guys, I was the one who introduced marker interfaces for WS actions :) and here it does not work very well for classes that define multiple actions. But still, I was planning on doing this refactoring (I have 15 more actions to add to this WS).